ACPT001.201: Test for SIO serial presence in ACPI tables

matmacieje opened the pull request:
Introducing dasharo-compatibility/acpi-yables.robot suite, requires mounted sysfs and platform config variables: ${SIO_NODE}, ${SIO_SCOPE}, ${SIO_SER_NODE}, ${SIO_SER_SCOPE}. Platform config set for Protectli V1x10 series as a referece.

DonVasyl committed:
ACPT001.201: Test for SIO serial presence in ACPI tables

Introducing dasharo-compatibility/acpi-tables.robot suite,
requires mounted sysfs and platform config variables: ${SIO_NODE},
${SIO_SCOPE}, ${SIO_SER_NODE}, ${SIO_SER_SCOPE}. Platform config
set for Protectli V1x10 series as a referece.

Signed-off-by: Mateusz Maciejewski <mateusz.maciejewski@3mdeb.com>

macpijan commented on platform-configs/include/protectli-v1x10.robot:
```diff
@@ -34,6 +34,11 @@ ${EXTERNAL_DISPLAY_PORT_SUPPORT}=               ${FALSE}
 
 ${DCU_SUPPORTED_BOOLEAN_SMMSTORE_VARIABLE}=     ${EMPTY}
 
+${SIO_NODE}=                                    PNP0A05:00
```
How platform-specific are these?

When we would add these to other Protectli boards?

mkopec replied:
Not specific at all, it's a generic PNP ID that applies to all boards with superios

macpijan replied:
So these can be defined in test suite and we simply enable this suite via a flag for all platforms with SuperIO chip?

Currently, there is not flag for this case, so it would be executed for all platforms when running regression.

matmacieje replied:
Changed in https://github.com/Dasharo/open-source-firmware-validation/pull/858/commits/bc7e032fd1efee2588a751b1e3eb4d6ed877dfb2

macpijan replied:
How about the flag then?

Now when running regression.sh, it would be also executed on e.g. laptops, where this tests may never pass?

matmacieje replied:
Flag introduced in https://github.com/Dasharo/open-source-firmware-validation/pull/858/commits/9330fd9b55b95b394335e00acd69395451fa1b14

[MSI_PRO_UBUNTU_acpi-tables_2025_06_10_17_19_12.zip](https://github.com/user-attachments/files/20675684/MSI_PRO_UBUNTU_acpi-tables_2025_06_10_17_19_12.zip)
[V1610_UBUNTU_acpi-tables_2025_06_10_17_17_34.zip](https://github.com/user-attachments/files/20675685/V1610_UBUNTU_acpi-tables_2025_06_10_17_17_34.zip)

macpijan commented on dasharo-compatibility/acpi-tables.robot:
```diff
@@ -0,0 +1,43 @@
+*** Test Cases ***
+ACPT001.201 SuperIO UART presence in sysfs ACPI tree (Ubuntu)
+    [Documentation]    Test verifies presence of SIO ACPI entries in sysfs.
+    ...    Platform config must contain SIO_NODE, SIO_SER_NODE, SIO_SCOPE
```
Shouldn't we check for these in tests then and produce informative error message?

matmacieje replied:
Removed in https://github.com/Dasharo/open-source-firmware-validation/pull/858/commits/bc7e032fd1efee2588a751b1e3eb4d6ed877dfb2

macpijan commented on platform-configs/include/protectli-v1x10.robot:
```diff
@@ -34,6 +34,11 @@ ${EXTERNAL_DISPLAY_PORT_SUPPORT}=               ${FALSE}
 ${DCU_SUPPORTED_BOOLEAN_SMMSTORE_VARIABLE}=     ${EMPTY}
 
+${SIO_NODE}=                                    PNP0A05:00
+${SIO_SCOPE}=                                   \\_SB_.PCI0.LPCB.SIO0
```
Also, when do we add new variables to https://github.com/Dasharo/open-source-firmware-validation/blob/develop/platform-configs/include/default.robot and when not?

matmacieje replied:
Moved to test suite in https://github.com/Dasharo/open-source-firmware-validation/pull/858/commits/bc7e032fd1efee2588a751b1e3eb4d6ed877dfb2

DonVasyl committed:
acpi-tables.robot: SuperIO SIO_ variables moved to test suite.

SIO_NODE, SIO_SCOPE, SIO_SER_NODE, SIO_SER_SCOPE moved from
protectli-v1x10.robot to test suite Variables section.

Signed-off-by: Mateusz Maciejewski <mateusz.maciejewski@3mdeb.com>

matmacieje commented:
Log from execution after https://github.com/Dasharo/open-source-firmware-validation/pull/858/commits/bc7e032fd1efee2588a751b1e3eb4d6ed877dfb2

[V1610_UBUNTU_acpi-tables_2025_06_10_15_49_43.zip](https://github.com/user-attachments/files/20674070/V1610_UBUNTU_acpi-tables_2025_06_10_15_49_43.zip)

DonVasyl committed:
HAS_SUPERIO_SERIAL flag introduced.

Signed-off-by: Mateusz Maciejewski <mateusz.maciejewski@3mdeb.com>

mkopec commented:
Needs a rebase

matmacieje commented:
@mkopec done, results after rebase: 
[UBUNTU_acpi-tables_2025_06_12_11_43_04.zip](https://github.com/user-attachments/files/20706438/UBUNTU_acpi-tables_2025_06_12_11_43_04.zip)
